### DISCUSSION TITLE
Is it possible to set a single image/logo for all social cards? Or can we disable them completely? (without the plugin?) [Windows trouble]

### DISCUSSION THREAD

**Discussion by MaximilianKohler** [squidfunk/mkdocs-material#5761] [Q&A] [answered]
I'm on Windows and having trouble setting up the social cards. Right now, websites are using the first image from each page. I'd like to either disable them from pulling any image, or set it to be a single image for all pages (**Edit**: [solution using overrides](https://github.com/squidfunk/mkdocs-material/discussions/5761#discussioncomment-6554599)). 

Regarding my problems:
I saw on https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/ that there's a video guide. Unfortunately it's for MacOS. 

I think I'm stuck here https://squidfunk.github.io/mkdocs-material/setup/dependencies/image-processing/#windows on the `pngquant` directions, and I opened an issue there about it https://github.com/jibsen/pngquant-winbuild/issues/6. 

I tried both of these, but neither disabled the previews completely:

```
plugins:
  - social:
      cards: false
      enabled: false
```
In `index.md`:
```
---
title: mywebsite
description: mywebsite description
social:
  cards: false
---
```

When I build on Windows with
```
plugins:
  - social
```
or anything similar, I get these errors which I haven't been able to solve:

```
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\Scripts\mkdocs.exe\__main__.py", line 7, in <module>
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1395, in invoke
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\mkdocs\__main__.py", line 250, in build_command
    build.build(cfg, dirty=not clean)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\mkdocs\commands\build.py", line 270, in build
    config = config.plugins.run_event('config', config)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\mkdocs\plugins.py", line 520, in run_event
    result = method(item, **kwargs)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\material\plugins\social\plugin.py", line 116, in on_config
    self.font = self._load_font(config)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\material\plugins\social\plugin.py", line 398, in _load_font
    self._load_font_from_google(name)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\material\plugins\social\plugin.py", line 422, in _load_font_from_google
    zip = ZipFile(tmp)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\zipfile.py", line 1269, in __init__
    self._RealGetContents()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\zipfile.py", line 1336, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

I'm using Python 3.10.11. I installed the precompiled GTK runtime. And I had no issues with `pip install pillow cairosvg`. 

**Edit**: more info in addition to the marked solution: https://github.com/squidfunk/mkdocs-material/discussions/5761#discussioncomment-6553639

**Comment by kamilkrzyskow** [comment]
The post is a bit all over the place or it's just me being stupid, as I'm not sure what are you trying to achieve. 
Firstly, are you a sponsor of MkDocs Material to get access to the newer Social plugin with elaborate customization options? From the post I'm assuming yes, since you wanted to install pngquant too, which is needed for the optimization plugin, which is also Insiders only. I'm not a part of Insiders therefore, knowledge about those features is limited.

> I'm on Windows and having trouble setting up the social cards. Right now, websites are using the first image from each page. I'd like to either disable them from pulling any image, or set it to be a single image for all pages.

Here you tell that the plugin works since it generates the images based on some criteria, but later you talk about the error, which prevents you from continuing? The `social: cards: false` setting in `mkdocs.yml` disables cards creation altogether, and I never used the setting in the front matter at the top of the file.
As for the video guide, even if it's for MacOS, apart of the installation guide nothing else should differ.
Here is one of the first videos showcasing the new social plugin customization features including setting an image in the background: https://twitter.com/i/status/1654832324272431104

As for the error, according to the traceback the error at the bottom originated in the `_load_font_from_google` function, which downloads the fonts specified in `mkdocs.yml`. The issue could be that the font you've chosen isn't in Google Fonts, you made a typo etc., Maybe the code doesn't handle network errors, as in google prevents you from downloading the fonts and instead of the ZIP file you download a 404 page or something. 

I agree that the issue here is hard to pin point. However, maybe the discussion will lead to some sort of result 🤷

You could even try to reinstall mkdocs-material. Something along the lines of (for the non-Insiders version): 
```shell
pip install mkdocs-material --force-reinstall
```

EDIT:
I also see you're using the Python version from the Windows Store. My preference is to install it from the official website and control every aspect of the installation etc. then use `pip -m venv venv` to create Virtual Environments and so on. As I don't know how Python from the Windows Store run scripts (maybe some sort of sandbox etc.), it could be yet another possible cause for the corrupted Google Fonts zip file, but I don't think that's the case.

**Reply by kamilkrzyskow** [reply]
What I forgot to mention is that, since you reached the google fonts step, it means that you've installed the cairosvg dependency correctly. Also I don't think that whatever you're doing is related to Windows... but if nothing works you can try using GitHub Actions as a Linux workstation (a bit slow and has a time limit etc., but enough to try things out), or install WSL2 or use Codespaces. There are many ways to avoid Windows limitations nowadays.

**Reply by kamilkrzyskow** [reply]
I tried running mkdocs with this config:
```
theme:
  name: material
  font:
    text: X X X X X X
```
And I got the same error as you, therefore I feel stronger about the "Font isn't in Google Fonts, therefore it crashes" idea.

**Reply by MaximilianKohler** [reply]
>Firstly, are you a sponsor of MkDocs Material to get access to the newer Social plugin with elaborate customization options? From the post I'm assuming yes, since you wanted to install pngquant too, which is needed for the optimization plugin, which is also Insiders only. 

No, I'm not a sponsor and I have no idea what I'm doing. I'm just following the directions that exist. I was trying to install `pngquant` because it was listed as a dependency. I didn't see any info about what is required for "normal vs insiders". 

>Here you tell that the plugin works since it generates the images based on some criteria, but later you talk about the error, which prevents you from continuing?

No plugin is needed for websites to use the first image of each page. They do that automatically. 

>As for the video guide, even if it's for MacOS, apart of the installation guide nothing else should differ.

Yes, but the installation is where I'm stuck. 

For fonts, I have this:
```
  font:
    text: Noto Sans N'Ko
    code: Roboto Mono
```
>My preference is to install it from the official website and control every aspect of the installation etc. then use pip -m venv venv to create Virtual Environments and so on.

I'm a noob, that sounds more complicated.

**Reply by MaximilianKohler** [reply]
I went with your "google font error" hint, and I changed to:
```
  font:
    text: Times New Roman
    code: Roboto Mono
```
But I still got the error when using:
```
plugins:
  - search:
      separator: '[\s\-,:!=\[\]()"`/]+|\.(?!\d)|&[lg]t;|(?!\b)(?=[A-Z][a-z])'
  - social
```

**Reply by kamilkrzyskow** [reply]
OK, so based on the code from the function:
https://github.com/squidfunk/mkdocs-material/blob/10b150112ea7b0e894419ce3351c17ac9e8e0cd7/material/plugins/social/plugin.py#L498-L499
Your final URL should look like this `https://fonts.google.com/download?family=Noto+Sans+N'Ko` which results in a 404 page.
After searching for the font in Google Fonts, the file name is `Noto_Sans_NKo.zip`, which means that the `'` symbol should be omitted. The plugin should clean the name from such characters, likely a bug.

Try with `text: Noto Sans NKo`

> No plugin is needed for websites to use the first image of each page. They do that automatically.

Without a plugin no social cards are generated for the page. Social cards are the images shown when you share the link on social media. I saw on your profile that you have a website called Human Microbiome built without the Social plugin with Material 9.0. 
If you send a link from that website onto Discord, Twitter or Facebook it shouldn't show any image or at best it would show an enlarged logo (theme: faviocon).
So having said that, I don't know what do you mean with "use the first image of each page automatically"

**Reply by kamilkrzyskow** [reply]
I missed that when writing the above message:
> But I still got the error when using: 
> ```
> plugins:
>   - search:
>       separator: '[\s\-,:!=\[\]()"`/]+|\.(?!\d)|&[lg]t;|(?!\b)(?=[A-Z][a-z])'
>   - social
> ```

I don't see the connection here... the traceback was exactly the same?

**Reply by MaximilianKohler** [reply]
>So having said that, I don't know what do you mean with "use the first image of each page automatically"

Look at https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fhumanmicrobiome.info%2F for example. You can see it uses the first image on the page. Same for https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fhumanmicrobiome.info%2Ffmt, etc.

>I don't see the connection here... the traceback was exactly the same?

I'm not sure what you mean by `traceback`. Oh, I see the "error output" includes the word "traceback", so here's the output:

```
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\Scripts\mkdocs.exe\__main__.py", line 7, in <module>
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\click\core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\mkdocs\__main__.py", line 250, in build_command
    build.build(cfg, dirty=not clean)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\mkdocs\commands\build.py", line 270, in build
    config = config.plugins.run_event('config', config)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\mkdocs\plugins.py", line 520, in run_event
    result = method(item, **kwargs)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\material\plugins\social\plugin.py", line 116, in on_config
    self.font = self._load_font(config)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\material\plugins\social\plugin.py", line 398, in _load_font
    self._load_font_from_google(name)
  File "C:\Users\User1\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\material\plugins\social\plugin.py", line 422, in _load_font_from_google
    zip = ZipFile(tmp)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\zipfile.py", line 1269, in __init__
    self._RealGetContents()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\lib\zipfile.py", line 1336, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

I compared them in notepad++ and the only difference I see is this new bit `    return ctx.invoke(self.callback, **ctx.params)` but perhaps I somehow left that out of the previous copy-paste.

**Reply by kamilkrzyskow** [reply, marked as answer]
Actually, I assumed you're correct, but just checked and there is no 'Times New Roman' font in Google Fonts either. Try with what I posted above or simply `text: Roboto`

~~As for the Linkedin, I will check it later once I find my account...~~ https://github.com/squidfunk/mkdocs-material/discussions/5761#discussioncomment-6554369

The message below contains more information if this answer together with the previous posts isn't enough:
https://github.com/squidfunk/mkdocs-material/discussions/5761#discussioncomment-6553639

**Reply by squidfunk** [reply]
System fonts are currently not supported by the social plugin, because they differ among systems.

**Reply by MaximilianKohler** [reply]
Interesting, `text: Roboto` or `text: Noto Sans` both worked. 

Wow, I was thinking that setting a single image for all pages would be preferable but I see the generated images in `.cache\plugin\social` and they look quite nice! Nice @squidfunk!

Just to confirm though, there is no option to set a **single static image for all pages**?

One issue I see is that the main heading text gets cut off instead of being shrunk to a smaller size https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fhumanmicrobiome.info%2Ffmt

Hmm, even the description gets cut off https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fhumanmicrobiome.info%2F

Ah, I see there's an experimental insiders option to shrink it https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#overflow

**EDIT**: `    text: PT Sans Narrow` _is a good font to use since it's more compact and decreases the need for shrinking_. 

Oh, I see the front matter option I tried before is insiders-only https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#disabling-social-cards
```
---
title: my page title
description: my page description
social:
  cards: false
---
```

Hmm, strange... I set it back to `text: Noto Sans N'Ko` and it still worked. Perhaps due to it being cached? https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#caching

**Reply by kamilkrzyskow** [reply]
Why did you select your message as the answer when I was the one who said it first... 🙄 Anyways, did you try the `text: Noto Sans NKo` setting, as this is the font without the special character, so it should be able to download it.

The change in the font should reset the cache, as every image needs to be redone with the new font, so I don't know why it worked with the font containing the special character 🤔 Unless this is yet another bug in the plugin, currently away from the PC so I don't want to check the code.

EDIT2: So I got to the PC and checked:
https://github.com/squidfunk/mkdocs-material/blob/3e8faa0fdba72454f0297d06c21bbde75baa10a2/material/plugins/social/plugin.py#L207-L213
The cache does indeed NOT take into account the font name, and neither the hash of the logo. I guess that the TODO comment has been forgotten😅

**Reply by MaximilianKohler** [reply]
>Why did you select your message as the answer when I was the one who said it first...

Because it contained more info (_including the fact that I switched back and I still don't get the error_), but I changed to your comment. I appreciate your help very much. 

>Anyways, did you try the text: Noto Sans NKo setting, as this is the font without the special character, so it should be able to download it.

No, I mentioned in the last comment that I set it back to `text: Noto Sans N'Ko` and it still worked.

**Reply by kamilkrzyskow** [reply]
> > So having said that, I don't know what do you mean with "use the first image of each page automatically"
> 
> Look at https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fhumanmicrobiome.info%2F for example. You can see it uses the first image on the page. Same for https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fhumanmicrobiome.info%2Ffmt, etc.

OK, I checked both the LinkedIn Post Inspector and Facebook Sharing Debuger https://developers.facebook.com/tools/debug/ 
It's indeed as you've said, that those sites load the first image that appears on the site if there is no `<meta>` tag pointing to a social card.

Either it changed over the years, or I just never noticed that I always shared sad pages without images, therefore I wrongly learned no image is loaded. Thanks for pointing me into the right direction.

**Reply by kamilkrzyskow** [reply]
> > Anyways, did you try the text: Noto Sans NKo setting, as this is the font without the special character, so it should be able to download it.
> 
> No, I mentioned in the last comment that I set it back to `text: Noto Sans N'Ko` and it still worked.

It didn't regenerate the images to use that font, just loaded them from the cache. I asked about `Noto Sans NKo`, since you mentioned `Noto Sans` without the `NKo`. The issue might resurface until it's fixed in the code. I will see if I can make those fixes.

@squidfunk For note, if I forget and disappear for weeks, @MaximilianKohler found 3 issues with the Social plugin:
- `_load_font_from_google` crashes with "invalid zip file" info when the font wasn't found in Google Fonts (so instead of the zip it downloads the 404 page)
- The URL to the font using his font `Noto Sans N'Ko` is wrong leading again to 404. I don't know if it's universal, but in this case the `'` symbol should be removed from the string to create the correct URL, at least I think that's the correct link because the file name is the same.
- The image cache doesn't take the name of font into consideration, also doesn't take the contents of the logo itself into consideration. So any font change or logo change won't update the cached images.

**Reply by MaximilianKohler** [reply]
>there is no `<meta>` tag pointing to a social card

So there's a way for us to set a `<meta>` tag (_for each/all pages_?) that points to one image file that can be used for all pages (_instead of the social plugin that generates a unique image for each page_)?

**Reply by kamilkrzyskow** [reply]
> > there is no `<meta>` tag pointing to a social card
> 
> So there's a way for us to set a `<meta>` tag (_for each/all pages_?) that points to one image file that can be used for all pages (_instead of the social plugin that generates a unique image for each page_)?

Yes, you can make use of the [override](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks) functionality, and override the `extrahead` block. The `<meta>` tags used by the plugin are here: 

https://github.com/squidfunk/mkdocs-material/blob/3e8faa0fdba72454f0297d06c21bbde75baa10a2/material/plugins/social/plugin.py#L368-L388

So your `main.html` override would look like this:

```html
{% extends "base.html" %}

{% block extrahead %}
<meta property="og:type" content="website">
<meta property="og:title" content="Microbiome Research Library">
<meta property="og:description" content="Library of research on the human microbiome. Understanding the influence of the gut microbiome on our well-being, and of the environment on our microbiome">
<meta property="og:image" content="https://humanmicrobiome.info/assets/images/social/index.png">
<meta property="og:image:type" content="image/png">
<meta property="og:image:width" content="1200">
<meta property="og:image:height" content="630">
<meta property="og:url" content="https://humanmicrobiome.info/">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:title" content="Microbiome Research Library">
<meta name="twitter:description" content="Library of research on the human microbiome. Understanding the influence of the gut microbiome on our well-being, and of the environment on our microbiome">
<meta name="twitter:image" content="https://humanmicrobiome.info/assets/images/social/index.png">
{% endblock %}
```

> Notice that there are repeating values.

The resolution 1200x630 is the most optimal, so best not to change and adjust your own image accordingly.

You can also make it a bit more dynamic based on the `mkdocs.yml`. Instead of hardcoding the content of the description into the `main.html` you can instead use `"{{ config.site_description }}"`, then it would load the description from the `mkdocs.yml` in case you'd change it etc. Similar stuff can be made with the rest of the values, so you can ask if you need more guidance.

**Reply by squidfunk** [reply]
> The cache does indeed NOT take into account the font name, and neither the hash of the logo. I guess that the TODO comment has been forgotten😅

@kamilkrzyskow it wasn't forgotten – this was all solved as part of the new version of the social plugin, which is in fact a ground-up rewrite. The new social plugin detects changes in all input data that is used to generate a social card and even deduplicates layers that stay the same between cards (e.g. the logo) to speed up the process of generation.

Other than that, thanks for sharing the overrides to set a single social card image! This is in fact beyond the scope of the plugin and can easily be achieved with overrides. Note that setting a single social card image doesn't convert (into users visiting your site) as well as generating specific images for each page, which is the raison d'être of the social plugin 😊

**Comment by MaximilianKohler** [comment]
I [upgraded MKdocs](https://www.mkdocs.org/about/release-notes/#upgrading) and [Material](https://squidfunk.github.io/mkdocs-material/upgrade/) today and noticed that you "[fixed the bug](https://squidfunk.github.io/mkdocs-material/changelog/#9.1.20)". Unfortunately, for me this caused more problems:

I was building with the Google Font I wanted for the social cards, and then changing the font to `Verdana` and using that on my site. I could leave it on `Verdana` and it wouldn't cause any problems, nor would it change the social card font. 

Unfortunately, after the update it will now not allow me to build with `Verdana` at all. I looked at the [Overrides](https://github.com/squidfunk/mkdocs-material/discussions/5761) section, but do not understand how I might be able to bypass this. Perhaps `\docs\stylesheets\extra.css` could be used as well? But I'm not sure how. 

Also, it only seems to be a "partial fix" since the .cache folder/files don't change at all when I change the font in `mkdocs.yml`. I don't think I'd want that "fixed" though. 

## Downgrading:

I don't see any info on downgrading to previous versions: https://squidfunk.github.io/mkdocs-material/ - via search for `downgrade`. 

Nor in the `issues` https://github.com/squidfunk/mkdocs-material/issues?q=is%3Aissue+downgrade

Nor in `discussions`: https://github.com/squidfunk/mkdocs-material/discussions?discussions_q=downgrade

---

I guess the "best" solution might be to find a google font similar to Verdana, but I haven't found one yet. I switched to Verdana from `Noto Sans N'Ko` because I noticed that N'Ko wasn't distinguishing "I" from lowercase "l", like it shows on the google page https://fonts.google.com/?preview.text=Lowercase%20l%20shouldn%27t%20look%20like%20I&preview.text_type=custom&query=Noto+Sans+N%27Ko

---

**EDIT**: 
`Noto Sans Vithkuqi` or `Noto Sans Adlam Unjoined` seem to work well enough.

And maybe this css would work:
```
:root {
  --md-text-font: "Verdana";
  --md-code-font: "Roboto Mono";
}
```

**Reply by kamilkrzyskow** [reply]
You linked to the 9.1.20/21 version update, the changes related to the Social plugin did not change the user exposed functionality at all. 
I think that the Social plugin fonts should match the font used on the page, and therefore the cache should be reset after the font change IMO, so once I get to "fixing" I might propose such change in my PR. 
You seem to have a special case where that doesn't work out, I'm not sure if you're creating the problem or doing something wrong 🤔 

As for "downgrading" I think you want to install a previous version of mkdocs-material, right? You can do that via `pip install mkdocs-material==9.1.0` where you specify the version, you can also add the `--force-reinstall` option to make sure everything is "freshly installed"

How to override the font via css is briefly described here: 
https://squidfunk.github.io/mkdocs-material/setup/changing-the-fonts/#customization
but I never done that, so I'm not sure whether or not you're supposed to add the font files in your docs deployment, but probably that's the case.

**Reply by squidfunk** [reply]
Setting a font family via CSS has no effect on the social plugin. As noted in my previous comment, it currently must be a Google Font. However, we can add support for custom TrueType and OpenType fonts to the new social plugin in the future 😊

**Reply by MaximilianKohler** [reply]
> Setting a font family via CSS has no effect on the social plugin. As noted in my previous comment, it currently must be a Google Font. 

Yes, my goal was to have Verdana as the main text on my website, and a Google Font for the social cards. CSS would work fine for that right? (IE: Google font in `mkdocs.yml` and Verdana in `extra.css`)

**Reply by squidfunk** [reply]
Jup, that's right.

**Reply by MaximilianKohler** [reply]
>I think that the Social plugin fonts should match the font used on the page, and therefore the cache should be reset after the font change IMO, so once I get to "fixing" I might propose such change in my PR.

If you do, I would suggest making it an optional on/off switch _(and mentioned on the [social cards page](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/))_. One reason not mentioned yet is that it takes longer to build/serve, and if a person is just experimenting with other things then it would be an unnecessary time-consumption. Also, some fonts download large files to the `.cache` which may also be unwanted.

**Reply by MaximilianKohler** [reply]
> Yes, my goal was to have Verdana as the main text on my website, and a Google Font for the social cards. CSS would work fine for that right? (IE: Google font in `mkdocs.yml` and Verdana in `extra.css`)

Hmm, this actually doesn't seem like a valid "fix". I just noticed that mobile browsers don't seem to care about the CSS, and they revert to their default font (not even the one in `mkdocs.yml`).

**Reply by squidfunk** [reply]
> > Yes, my goal was to have Verdana as the main text on my website, and a Google Font for the social cards. CSS would work fine for that right? (IE: Google font in `mkdocs.yml` and Verdana in `extra.css`)
> 
> Hmm, this actually doesn't seem like a valid "fix". I just noticed that mobile browsers don't seem to care about the CSS, and they revert to their default font (not even the one in `mkdocs.yml`).

Then you likely have an error in your CSS file or the CSS file was not loaded correctly.
